File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int minLength = numeric_limits<int>::max();
        int index = -1;
        string prefix = "";
        int lengthArray = strs.size();
        if(lengthArray==1){
            return strs[0];
        }
        for(int i = 0;i<lengthArray;i++){
            if(strs[i].size()<minLength){
                minLength = strs[i].size();
                index = i;
            }
        }
        string firstWord = strs[index];
        int lengthFirstWord = firstWord.size();
        for(int i = 0;i<lengthFirstWord;i++){
            char currChar = firstWord.at(i);
            for(int j = 0;j<lengthArray;j++){
                if(strs[j].at(i)!=currChar){
                    return prefix;
                }
            }
            prefix += currChar;
        }
        return firstWord;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int min = Integer.MAX_VALUE;
        String smallest = "";
        for(int i = 0;i<strs.length;i++){
            int l = strs[i].length();
            if(l<min){
                min = l;
                smallest = strs[i];
            }
        }
        List<Character> prefix = new ArrayList<>();
        int signal = -1;
        for(int i = 0;i<min;i++){
            char temp = smallest.charAt(i);
            for(int j = 0;j<strs.length;j++){
                char tempy = strs[j].charAt(i);
                if(tempy!=temp){
                    j=strs.length;
                    i = min;
                    signal = 1;
                }
            }
            if(signal<0){
                            prefix.add(temp);

            }
        }
        String result = "";
        for(int i = 0;i<prefix.size();i++){
            result = result+prefix.get(i);
        }
        return result;
    }
}